### PR TITLE
feat: add cloudsql provider

### PR DESCRIPTION
**Pull request by Yuan325**
cloudsql-python-connector will need to use sqlalchemy for connection pooling.

### CODE REVIEW

File: extension_service/datastore/providers/cloudsql.py [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+                      embedding vector(768) NOT NULL
+                    )
+                    """
+                )
+            )
+            # Insert all the data
+            await conn.execute(
+                sqlalchemy.text(
+                    """INSERT INTO amenities VALUES (:id, :name, :description, :location, :terminal, :category, :hour, :content, :embedding)"""
+                ),
```
Comment (kurtisvg) [inline extension_service/datastore/providers/cloudsql.py:144]: super-nit: we're using `sqlalchemy.text` enough it's probably cleaner to `from sqlalchemy import text` instead

File: extension_service/datastore/providers/cloudsql.py [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+                    """
+                        ),
+                        parameters={
+                            "id": id,
+                        },
+                    )
+                )
+                .mappings()
+                .fetchone()
+            )
```
Comment (kurtisvg) [inline extension_service/datastore/providers/cloudsql.py:329]: don't be afraid to not nest, especially on complicated/long statements like this -- it often improves readability to pull it out.
```suggestion
            s = text(
                """
                SELECT id, name, description, location, terminal, category, hour
                FROM amenities WHERE id=:id
                """
            )
            p = {  "id": id }
            result = await conn.execute(s, parameters=p).mappings().fetchone()
```

File: extension_service/pyproject.toml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 profile = "black"
 
 [tool.mypy]
-python_version = 3.10
+python_version = "3.10"
```
Comment (kurtisvg) [inline extension_service/pyproject.toml:5]: nit: these should probably be fixed in separate PRs

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: extension_service/datastore/providers/__init__.py [hunk 4]
```diff
@@ -12,6 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from . import postgres
+from . import cloudsql, postgres
 
-__ALL__ = [postgres]
+__ALL__ = [postgres, cloudsql]
```
[the remark this answers is not in the document]
Comment (kurtisvg) [reply extension_service/datastore/providers/__init__.py:17]: `cloudsql_postgres`

File: extension_service/datastore/providers/cloudsql.py [hunk 5]
```diff
@@ -0,0 +1,464 @@
...
+                        "name": a.name,
+                        "description": a.description,
+                        "location": a.location,
+                        "terminal": a.terminal,
+                        "category": a.category,
+                        "hour": a.hour,
+                        "content": a.content,
+                        "embedding": None
+                        if a.embedding is None
+                        else "[" + ",".join(str(e) for e in a.embedding) + "]",
```
[the remark this answers is not in the document]
Comment (Yuan325) [reply extension_service/datastore/providers/cloudsql.py]: This worked! Thankyou!

**Review by kurtisvg — APPROVED** [review]
